shell/utils/unit-tests/ChildRenderingRouterLinkStub.ts

```diff
@@ -0,0 +1,36 @@
+import { RouterLinkStub } from '@vue/test-utils';
+import { NavigationFailure, Route } from 'vue-router';
+
+/**
+ * See {@link RouterLinkSlotArgument} in vue-router
+ */
+export interface RouterLinkSlotArgumentOptional {
+    href?: string;
+    route?: Route;
+    navigate?: (e?: MouseEvent) => Promise<undefined | NavigationFailure>;
+    isActive?: boolean;
+    isExactActive?: boolean;
+}
+
+/**
+ * This is a workaround because VueUtils RouterLinkStub doesn't currently support the slot api.
+ *
+ * See @link https://github.com/vuejs/vue-test-utils/issues/1803#issuecomment-940884170
+ *
+ * @param slotProps Provide arguments that you want passed to the child rendered by router-link
+ * @returns A stub
+ */
+export function createChildRenderingRouterLinkStub(slotProps?: RouterLinkSlotArgumentOptional): typeof RouterLinkStub | any {
+  return {
+    ...RouterLinkStub,
+    render() {
+      return this.$scopedSlots.default({
```

Comment:
Can we use normal slots instead of scoped?

Reply:
No, it's empty.

![image](https://github.com/rancher/dashboard/assets/55104481/89761692-b30a-4cf8-85bc-87122f6901e4)